lib/news_scraper/trainer/uri_trainer.rb

```diff
@@ -1,57 +1,74 @@
+require 'pry'
+
 module NewsScraper
   module Trainer
     class UriTrainer
       def initialize(uri)
-        @article_scrape_patterns = YAML.load_file(Constants::SCRAPE_PATTERN_FILEPATH)
         uri_parser = URIParser.new(uri)
         @uri = uri_parser.without_scheme
         @root_domain = uri_parser.host
-        @raw_html = Extractors::Article.new(uri: @uri).extract
+        @payload = Extractors::Article.new(uri: @uri).extract
       end
 
       def train
-        unless @article_scrape_patterns['domains'].key?(@root_domain)
-          CLI.put_header
-          CLI.log("There is no scrape pattern defined for #{@root_domain}"\
-            " in config/article_scrape_patterns.yml")
-          no_scrape_defined
+        return if article_scrape_patterns['domains'].key?(@root_domain)
+
+        CLI.put_header(@root_domain)
+        CLI.log("There is no scrape pattern defined for #{@root_domain} in config/article_scrape_patterns.yml")
+        CLI.log "Fetching information..."
+        CLI.put_footer
+
+        selected_presets = {}
+        article_scrape_patterns['data_types'].each do |data_type|
+          selected_presets[data_type] = pattern(data_type)
         end
+
+        save_selected_presets(selected_presets)
       end
 
-      def no_scrape_defined
-        if CLI.confirm("Step through presets for #{@root_domain}?")
-          CLI.put_footer
-          selected_presets = Trainer::PresetsSelector.new(uri: @uri).select(@raw_html)
+      private
 
-          CLI.put_header('Save preset')
-          CLI.log_lines(selected_presets.to_yaml)
-          if CLI.confirm("Save these scrape patterns for #{@root_domain}?")
-            save_selected_presets(selected_presets)
-          end
+      def pattern(data_type)
+        CLI.put_header("Determining information for #{data_type}")
+        data_type_presets = article_scrape_patterns['presets'][data_type]
+        pattern = if data_type_presets.nil? || data_type_presets.empty?
```

Comment:
Checking for `empty?` is too defensive IMO. We'll never add a `data_type` to the YAML file without a preset.